CMakeLists.txt

```diff
@@ -8,51 +8,96 @@ set(CMAKE_TOOLCHAIN_FILE ${CMAKE_SOURCE_DIR}/cmake/stm32-toolchain.cmake)
 project(ATeam-Firmware
 	LANGUAGES C CXX ASM)
 
+macro(tests dir linker fname tcomment prefix)
```

Comment:
maybe move this macro to its own file cmake/ and then include it? or it could go in cmake/stm32-defines.cmake